proc/rename/rename.go

```diff
@@ -16,71 +17,77 @@ import (
 // applied left to right; each rename observes the effect of all
 // renames that preceded it.
 type Proc struct {
-	pctx       *proc.Context
-	parent     proc.Interface
-	fieldnames []string
-	targets    []string
-	typeMap    map[int]*zng.TypeRecord
+	pctx   *proc.Context
+	parent proc.Interface
+	// For the dst field name, we just store the leaf name since the
+	// src path and the dst path are the same and only differ in the leaf name.
+	dsts    []field.Static
+	srcs    []field.Static
+	typeMap map[int]*zng.TypeRecord
 }
 
 func New(pctx *proc.Context, parent proc.Interface, node *ast.RenameProc) (*Proc, error) {
-	var fieldnames, targets []string
+	var srcs, dsts []field.Static
 	for _, fa := range node.Fields {
-		ts := strings.Split(fa.Target, ".")
-		fs := strings.Split(ast.FieldExprToString(fa.Source), ".")
-		if len(ts) != len(fs) {
-			return nil, fmt.Errorf("cannot rename %s to %s", ast.FieldExprToString(fa.Source), fa.Target)
+		dst, err := expr.CompileLval(fa.LHS)
+		if err != nil {
+			return nil, err
 		}
-		for i := range ts[:len(ts)-1] {
-			if ts[i] != fs[i] {
-				return nil, fmt.Errorf("cannot rename %s to %s (differ in %s vs %s)", fa.Source, fa.Target, ts[i], fs[i])
+		// We call CompileLval on the RHS because renames are
+		// restricted to dotted field name expressions.
+		src, err := expr.CompileLval(fa.RHS)
+		if err != nil {
+			return nil, err
+		}
+		if len(dst) != len(src) {
+			return nil, fmt.Errorf("cannot rename %s to %s", src, dst)
+		}
+		for i := len(src) - 2; i >= 0; i-- {
```

Comment:
Would it be better to iterate forward, thus showing the first path component that differs rather than the last?